### PR TITLE
Reset password and Contact Form(Meg's, but I am PRing it)

### PR DESCRIPTION
**Pull request by Blaine-Mason**
Add Reset Password Page
Add Reset Password Functionality
Add Contact Page(Meg)
Future Purpose: Admin forgets password -> enters username -> email sent to address in the db -> admin clicks link -> resets password.

### CODE REVIEW

**Comment by jtanderson** [comment]
Looks like some non-trivial business logic happening and the commit messages do not inspire confidence. @jventura1738 take a look and test.

**Comment by jacobduncan00** [comment]
Need more professional commit messages

**Review by jventura1738 — COMMENTED** [review]
Not very clear how to get to the reset password page; would be easier with a button.

File: api/forms.py [hunk 1]
```diff
@@ -85,3 +87,30 @@ def listing_submit():
...
+
+# Route for submitting password reset:
+@forms.route('/reset-password-submit', methods=['POST'])
+def reset_pass_submit():
+    """Reset password submission route.
+    This function handles the submission of a password reset.
+    """
+    data = request.json
+    response = dict()
+    user = ListingsModel.query.get(data["username"])
```
Comment (jventura1738) [inline api/forms.py]: 1.  Stick to single quotes for consistency.
2. When I put my username, I get an ID error in the SQL query.

File: api/views.py [hunk 2]
```diff
@@ -92,3 +92,14 @@ def insert_internship():
...
+
+# Renders reset password form:
+@views.route('/login/reset-password', methods=['GET'])
+def reset_password():
+    """Reset Password page for users. (Pending Auth)
+    This function runs whenever the reset-password page ('/reset-password') is requested.
+        ex) -> localhost:5000/login/reset-password
+    Returns the reset password page: which renders reset.html (reset form)
+    """
+    return render_template('reset.html', page_title='Reset Password')
```
Comment (jventura1738) [inline api/views.py]: Not an error but calling the template "reset_password.html" would be more intuitive

**Review by jventura1738 — APPROVED** [review]
Contact form operational.

Future PRs: backend implementation and an easier way to access page

File: api/forms.py [hunk 3]
```diff
@@ -85,3 +87,30 @@ def listing_submit():
 
     response = {'status': 200}
     return response
+
+
+# Route for submitting password reset:
+@forms.route('/reset-password-submit', methods=['POST'])
```
Comment (jventura1738) [inline api/forms.py]: Should be a PUT request since it is modifying the database

File: api/forms.py (hunk 1 above)
Comment (jventura1738) [reply api/forms.py]: > 1. Stick to single quotes for consistency.
> 2. When I put my username, I get an ID error in the SQL query.

I took a look, its using my username as an ID in the query.

**Review by jventura1738 — COMMENTED** [review]
Main issues:
1. Reset password should be a PUT request
2. On my machine, reset password is broken, check comment on that

**Comment by jacobduncan00** [comment]
> Not very clear how to get to the reset password page; would be easier with a button.

Agree with justin here, I can add back that "forgot password" link on the signin form I had that links to this

File: api/forms.py (hunk 3 above)
Comment (jacobduncan00) [reply api/forms.py]: Agreed as we are changing information already in the DB, not adding new info

**Comment by Blaine-Mason** [comment]
Comments addressed and changes ready to be merged.

**Review by jventura1738 — APPROVED** [review]
Tested on my machine, works well and ready for merge @jtanderson.  

Have you already tested on yours @Blaine-Mason ?  I'm assuming you already have, but best to state you did just so we are confident it's been tested thoroughly.  Disregard if you have already done so

**Pull request merged**